frontend/ui-pablo/app/components/Organisms/bonds/BuyButtons/index.tsx

```diff
@@ -1,104 +1,99 @@
 import { BaseAsset, PairAsset } from "@/components/Atoms";
-import { getToken } from "@/defi/Tokens";
-import { BondDetails, TokenId } from "@/defi/types";
+import { SelectedBondOffer } from "@/defi/hooks/bonds/useBondOffer";
 import { Button, Grid, GridProps } from "@mui/material";
-import { useState } from "react";
-import { BuyLPTokenModal } from "./BuyLPTokenModal";
-import { BuySingleTokenModal } from "./BuySingleTokenModal";
+import { useRouter } from "next/router";
 
 const threeColumnPageSize = {
   xs: 12,
   sm: 12,
   md: 4,
 };
 
-const buttonProps = (onClick: () => void) => ({
-  variant: "outlined",
-  fullWidth: true,
-  onClick: onClick,
-} as const);
+const buttonProps = (onClick: () => void) =>
+  ({
+    variant: "outlined",
+    fullWidth: true,
+    onClick: onClick,
+  } as const);
 
-const restAssetProps = (label: string, iconSize: number) => ({
-  label: label,
-  LabelProps: {
-    variant: "body1",
-    fontWeight: "normal"
-  },
-  iconSize: iconSize,
-} as const);
+const restAssetProps = (label: string, iconSize: number) =>
+  ({
+    label: label,
+    LabelProps: {
+      variant: "body1",
+      fontWeight: "normal",
+    },
+    iconSize: iconSize,
+  } as const);
 
 type TokenType = "token1" | "token2" | "lp";
 
 export type BuyButtonsProps = {
-  bond: BondDetails;
+  bond: SelectedBondOffer;
   iconSize?: number;
 } & GridProps;
 export const BuyButtons: React.FC<BuyButtonsProps> = ({
   bond,
   iconSize = 24,
   ...gridProps
 }) => {
-  const token1 = getToken(bond.tokenId1);
-  const token2 = getToken(bond.tokenId2);
-
-  const [activeToken, setActiveToken] = useState<TokenType>("token1");
-  const [isOpenBuyModal, setIsOpenBuyModal] = useState<boolean>(false);
-
-  const isBuyLpToken = activeToken === "lp";
-  const tokenIdToBuy = activeToken === "token1" ? bond.tokenId1 : bond.tokenId2;
+  const router = useRouter();
 
   const onBuyHandler = (token: TokenType) => () => {
-    setActiveToken(token);
-    setIsOpenBuyModal(true);
+    if (token === "lp") {
+      router.push("/pool");
+    } else {
+      router.push("/swap");
+    }
   };
 
-  const onCloseBuyModal = () => {
-    setIsOpenBuyModal(false);
-  };
+  const isLpBond =
+    bond.principalAsset &&
+    (bond.principalAsset as any).baseAsset &&
+    (bond.principalAsset as any).quoteAsset;
+  if (!isLpBond) return null;
 
   return (
     <Grid container columnSpacing={3} {...gridProps}>
       <Grid item {...threeColumnPageSize}>
         <Button {...buttonProps(onBuyHandler("token1"))}>
           <BaseAsset
-            icon={token1.icon}
-            {...restAssetProps(token1.symbol, iconSize)}
+            icon={(bond.principalAsset as any).baseAsset.icon}
```

Comment:
This seems like some sort of type missmatch. Why is this conversion needed? - This comment applies to all such places - It might be good to rethink it